scripts/globals/npc_util.lua

```diff
@@ -7,25 +7,34 @@ npcUtil = {};
 -- Picks a new position for an NPC and excluding the current position.
 -- INPUT: npc = npcID, position = 2D table with coords: index, {x, y, z}
 -- RETURN: table index
-function npcUtil.pickNewPosition(npc, positionTable)
-
+function npcUtil.pickNewPosition(npc, positionTable, allowCurrentPosition)
     local npc = GetNPCByID(npc)
     local positionIndex = 1 -- Default to position one in the table if it can't be found.
     local tableSize = 0
+    local allowCurrentPosition = allowCurrentPosition or false;
+
     for i, v in ipairs(positionTable) do   -- Looking for the current position
 
-        -- Finding by comparing the NPC's coords
-        if (math.floor(v[1]) == math.floor(npc:getXPos()) and math.floor(v[2]) == math.floor(npc:getYPos()) and math.floor(v[3]) == math.floor(npc:getZPos())) then
-            positionIndex = i; -- Found where the NPC is!
+        if (allowCurrentPosition == "false") then
```

Comment:
`if (allowCurrentPosition == "ziggy") then` winds up being the same thing. Unless somewhere you are assigning the string to that variable, it can never be true. It only "works" because the comparison always returns false there. Since that variable is not that string. I don't think this is how you wanted that to work.